packages/block-editor/src/store/selectors.js

```diff
@@ -945,6 +946,21 @@ export function getBlockMode( state, clientId ) {
 	return state.blocksMode[ clientId ] || 'visual';
 }
 
+/**
+ * Determines if block mode toggle is allowed for a given block.
+ *
+ * @param {Object} state    Editor state.
+ * @param {string} clientId Block client ID.
+ *
+ * @return {boolean} Whether block mode toggle is allowed on given block.
+ */
+export function canToggleBlockMode( state, clientId ) {
```

Comment:
1. Added canToggleBlockMode selector, should I update docs? If so how to do that?
2. Is it possible to add isCodeEditingEnabled check logic (`const isCodeEditingEnabled = select( 'core/editor' ).getEditorSettings().codeEditingEnabled;`) to this selector? That would further remove duplication.